src/nm.c

```diff
@@ -226,6 +217,21 @@ write_routes(const NetplanNetDefinition* def, GString *s, int family)
             if (cur_route->metric != NETPLAN_METRIC_UNSPEC)
                 g_string_append_printf(s, ",%d", cur_route->metric);
             g_string_append(s, "\n");
+
+            if (   cur_route->onlink
+                || cur_route->table != NETPLAN_ROUTE_TABLE_UNSPEC
+                || cur_route->from) {
+                g_string_append_printf(s, "route%d_options=", j);
```

Comment:
I don't know the exact syntax of route_options=, so just want to make sure: does it accept multiple key=value pairs separated with spaces? So, let's say, "route1_options=onlink=true from=192.168.14.2" would be valid syntax?

Reply:
yes. In a slightly different syntax.


```
nmcli connection modify "$PROFILE" ipv4.routes \
     '192.168.14.0/24 onlink=1 src=192.168.14.2'
```
gives:

```
[ipv4]
route1=192.168.14.0/24
route1_options=onlink=true,src=192.168.14.2
```

 That's the only way to configure multiple route attributes for one route in keyfile format.

See https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/blob/7b9cb1f952e8c8a0001144f457fcea1c7c853218/shared/nm-keyfile/nm-keyfile.c#L547

Reply:
ah... then the patch is wrong :)

Sorry. Pushed a fixup commit...

Reply:
Ok, this looks good now, thanks! We could add a test-case for this I think, but I also don't consider it a blocker for getting this merged. Maybe something to follow up with later.